src/model.h

```diff
@@ -118,47 +115,30 @@ struct RapidsModel : rapids::Model<RapidsSharedState> {
 
     output.finalize();
 
-    /* For running gpu treeshap
-     * Make sure that input and output buffers are in device memory
-     * for gpu treeshap */
-    if constexpr (rapids::IS_GPU_BUILD && IS_TREESHAP_BUILD) {
-      // boolean to check whether gpu treeshap will be executed
-      auto run_gpu_treeshap =
-          shared_state->check_output_name("treeshap_output");
-
-      if (run_gpu_treeshap) {
-        auto treeshap_output = get_output<float>(batch, "treeshap_output");
-        auto treeshap_output_buffer = rapids::Buffer<float>(
-            treeshap_output.data(), treeshap_output.size(),
-            treeshap_output.mem_type(), treeshap_output.device(),
-            treeshap_output.stream());
-
-        if (gpu_treeshap_model.has_value()) {
-          // Always copy input buffer to device memory
-          if (input_buffer.mem_type() == rapids::HostMemory) {
-            input_buffer = rapids::Buffer<float const>(
-                input_buffer, rapids::DeviceMemory, get_device_id());
-          }
-
-          // This will force a copy of the output buffer to device memory
-          if (input_buffer.mem_type() != treeshap_output_buffer.mem_type()) {
-            // Create output buffer in correct  location
-            treeshap_output_buffer = rapids::Buffer<float>(
-                treeshap_output.size(), input_buffer.mem_type(),
-                get_device_id(), get_stream());
-          }
-
+    // boolean to check whether gpu treeshap will be executed
+    auto run_treeshap =
+        shared_state->check_output_name("treeshap_output");
```

Comment:
It looks like you've eliminated the `IS_TREESHAP_BUILD` option here. I'm actually fine with that (practically speaking, we never build without treeshap), but we should update the CMake infrastructure accordingly.